clang/lib/Analysis/CFG.cpp

```diff
@@ -4892,16 +4898,17 @@ CFGBlock *CFGBuilder::VisitCXXFunctionalCastExpr(CXXFunctionalCastExpr *E,
   return Visit(E->getSubExpr(), asc);
 }
 
-CFGBlock *CFGBuilder::VisitCXXTemporaryObjectExpr(CXXTemporaryObjectExpr *C,
+CFGBlock *CFGBuilder::VisitCXXTemporaryObjectExpr(CXXTemporaryObjectExpr *E,
                                                   AddStmtChoice asc) {
   // If the constructor takes objects as arguments by value, we need to properly
   // construct these objects. Construction contexts we find here aren't for the
   // constructor C, they're for its arguments only.
-  findConstructionContextsForArguments(C);
+  findConstructionContextsForArguments(E);
 
-  autoCreateBlock();
-  appendConstructor(Block, C);
-  return VisitChildren(C);
+  createConstructorBlock(E->getConstructor());
+  appendConstructor(Block, E);
```

Comment:
I do like the addition of `createConstructorBlock()`. Now that it exists, I noticed the only two calls to `appendConstructor()` are right after a call to `createConstructorBlock()`. I think we should just inline `appendConstructor()` into `createConstructorBlock()` at this point.

Reply:
@Sirraide Thanks. I've moved these changes to `appendConstructor`